src/query/v2/interpreter.hpp

```diff
@@ -296,13 +297,13 @@ class Interpreter final {
    */
   void Abort();
 
-  const RequestRouterInterface *GetRequestRouter() const { return request_router_.get(); }
+  RequestRouterInterface *GetRequestRouter() { return request_router_.get(); }
```

Comment:
Instead of exposing the whole non-const interface of `RequestRouter` from the `Interpreter`, I would keep the const getter and add a function to install the simulator ticker. That way the interpreter can steel keep a strict leash on the `RequestRouter`. With the current approach it is possible to call `interpreter.GetRequestRouter()->Commit()`, which is not desirable in my opinion.